apps/wallet-dashboard/components/Dialogs/Assets/AssetDialog.tsx

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState } from 'react';
+import { Dialog } from '@iota/apps-ui-kit';
+import { FormikProvider, useFormik } from 'formik';
+import { useCurrentAccount } from '@iota/dapp-kit';
+import { createNftSendValidationSchema } from '@iota/core';
+import { DetailsView, SendView } from './views';
+import { IotaObjectData } from '@iota/iota-sdk/client';
+import { AssetsDialogView } from './constants';
+import { useCreateSendAssetTransaction, useNotifications } from '@/hooks';
+import { NotificationType } from '@/stores/notificationStore';
+
+interface AssetsDialogProps {
+    onClose: () => void;
+    onSent?: () => void;
+    asset: IotaObjectData;
+}
+
+interface FormValues {
+    to: string;
+}
+
+const INITIAL_VALUES: FormValues = {
+    to: '',
+};
+
+export function AssetDialog({ onClose, onSent, asset }: AssetsDialogProps): JSX.Element {
+    const [view, setView] = useState<AssetsDialogView>(AssetsDialogView.Details);
+    const account = useCurrentAccount();
+    const activeAddress = account?.address ?? '';
+    const objectId = asset?.objectId ?? '';
+    const { addNotification } = useNotifications();
+    const validationSchema = createNftSendValidationSchema(activeAddress, objectId);
+
+    const { mutation: sendAsset } = useCreateSendAssetTransaction(objectId);
+
+    const formik = useFormik<FormValues>({
+        initialValues: INITIAL_VALUES,
+        validationSchema: validationSchema,
+        onSubmit: onSubmit,
+        validateOnChange: true,
+    });
+
+    async function onSubmit(values: FormValues) {
+        try {
+            await sendAsset.mutateAsync(values.to);
+            addNotification('Transfer transaction successful', NotificationType.Success);
+            onClose();
+            setView(AssetsDialogView.Details);
+            await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
why do we need this timeout here?

Reply:
Once we implement tx details, we wont close the dialog at this point but rather go to the next state which is the tx detail, and there if the tx is sucesful is where the refretch should happen, so the logic will be cohesive. I would remove this promise, we could even remove the refetch and it wont be needed at this step